### DISCUSSION TITLE
Why does the terraform provider need an SSH private key?

### DISCUSSION THREAD

**Discussion by alexklibisz** [kube-hetzner/terraform-hcloud-kube-hetzner#1205] [Q&A]
Hi, thanks for this incredible project. It has made spinning up k8s on Hetzner so much simpler.

I am curious about one thing: why do we need to provide an SSH private key, here: https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/blob/e911232e73951ec873711a388b58770e10b8a80f/kube.tf.example#L40?

I know it says to read the SSH docs (https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/blob/master/docs/ssh.md), but I also don't see any answer there.

Ideally, I would like to avoid providing my private key to any piece of software that I don't fully understand. For the time-being, I've just generated another key pair for use with this project.

**Comment by DjVinnii** [comment]
The SSH private key is being used to setup the SSH agent in order to remotely execute scripts on the nodes. Searching in the repo on `ssh_private_key` might give you a better overview on how this private key is exactly being used.

**Reply by alexklibisz** [reply]
Thanks, that's a bit clearer. What's the best practice for segregating this key? I'm assuming I should not use a key that I use for sshing to other machines I own. Should I create a separate key for each cluster that I build? (That's what I've done so far.)
